datasources/src/test/scala/com/normation/plugins/datasources/JsonPathTest.scala

```diff
@@ -109,28 +110,28 @@ class JsonPathTest extends Specification with BoxSpecMatcher with Loggable {
                 "title": "The Lord of the Rings",
                 "isbn": "0-395-19395-8",
                 "price": 22.99
-            }""").map(s => compactRender(parse(s))))
+            }""").map(s => GenericProperty.fromJsonValue(parse(s))))
     }
     "retrieve NUMBER childrens forming an array" in {
-      JsonSelect.fromPath("$.store.book[*].price", json).either.runNow must beRight(List("8.95", "12.99", "8.99", "22.99"))
+      JsonSelect.fromPath("$.store.book[*].price", json).either.runNow must beRight(List("8.95", "12.99", "8.99", "22.99").flatMap(p => GenericProperty.parseValue(p).toOption))
     }
     "retrieve STRING childrens forming an array" in {
-      JsonSelect.fromPath("$.store.book[*].category", json).either.runNow must beRight(List("reference", "fiction", "\"quotehorror\"", "fiction"))
+      JsonSelect.fromPath("$.store.book[*].category", json).either.runNow must beRight(List("reference", "fiction", "\"quotehorror\"", "fiction").map(p => GenericProperty.fromJsonValue(JString(p))))
```

Comment:
`GenericProperty.parseValue(...` - if it doesn't work, then there is something fishy about serialisation/escaping